kv/kvserver.TestPromoteNonVoterInAddVoter: TestPromoteNonVoterInAddVoter failed

kv/kvserver.TestPromoteNonVoterInAddVoter.TestPromoteNonVoterInAddVoter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20681543?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20681543?buildTab=artifacts#/) on master @ [98557a6304fe239582e20ebd07f69a3ee8e47bdf](https://github.com/cockroachdb/cockroach/commits/98557a6304fe239582e20ebd07f69a3ee8e47bdf):


```
=== RUN   TestPromoteNonVoterInAddVoter
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/135a86ba3501d232b0a98ae8902ff534/logTestPromoteNonVoterInAddVoter903965342
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:594: DRPC is enabled (override by TestingGlobalDRPCOption)
    replicate_queue_test.go:2283: 
        	Error Trace:	pkg/kv/kvserver/replicate_queue_test.go:2283
        	            				pkg/kv/kvserver/replicate_queue_test.go:2308
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:355
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/kv/kvserver/replicate_queue_test.go:2307
        	Error:      	Received unexpected error:
        	            	sql: no rows in result set
        	Test:       	TestPromoteNonVoterInAddVoter
    exectrace.go:39: execution trace written to /artifacts/tmp/_tmp/135a86ba3501d232b0a98ae8902ff534/logTestPromoteNonVoterInAddVoter903965342/exectrace_goid_89908.bin
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/135a86ba3501d232b0a98ae8902ff534/logTestPromoteNonVoterInAddVoter903965342
--- FAIL: TestPromoteNonVoterInAddVoter (47.29s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/kv-triage
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPromoteNonVoterInAddVoter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-56123

kv/kvserver.TestPromoteNonVoterInAddVoter.TestPromoteNonVoterInAddVoter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20690000?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/20690000?buildTab=artifacts#/) on master @ [0c7c0373ee7b8e7879ca918b95957c7784198d40](https://github.com/cockroachdb/cockroach/commits/0c7c0373ee7b8e7879ca918b95957c7784198d40):


```
=== RUN   TestPromoteNonVoterInAddVoter
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/3badd79dc476285a5b1104ad1a94c4a5/logTestPromoteNonVoterInAddVoter1152580899
    test_log_scope.go:82: use -show-logs to present logs inline
    test_server_shim.go:594: DRPC is enabled (override by TestingGlobalDRPCOption)
    replicate_queue_test.go:2283: 
        	Error Trace:	pkg/kv/kvserver/replicate_queue_test.go:2283
        	            				pkg/kv/kvserver/replicate_queue_test.go:2308
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:355
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/kv/kvserver/replicate_queue_test.go:2307
        	Error:      	Received unexpected error:
        	            	sql: no rows in result set
        	Test:       	TestPromoteNonVoterInAddVoter
    exectrace.go:39: execution trace written to /artifacts/tmp/_tmp/3badd79dc476285a5b1104ad1a94c4a5/logTestPromoteNonVoterInAddVoter1152580899/exectrace_goid_93141.bin
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/3badd79dc476285a5b1104ad1a94c4a5/logTestPromoteNonVoterInAddVoter1152580899
--- FAIL: TestPromoteNonVoterInAddVoter (36.79s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPromoteNonVoterInAddVoter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

kv/kvserver.TestPromoteNonVoterInAddVoter.TestPromoteNonVoterInAddVoter [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20697429?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsIbmCloudLinuxS390x_UnitTestsS390x/20697429?buildTab=artifacts#/) on master @ [8f19f413325301b52fb065940e6c067072b96098](https://github.com/cockroachdb/cockroach/commits/8f19f413325301b52fb065940e6c067072b96098):


```
=== RUN   TestPromoteNonVoterInAddVoter
    test_log_scope.go:171: test logs captured to: /artifacts/tmp/_tmp/135a86ba3501d232b0a98ae8902ff534/logTestPromoteNonVoterInAddVoter844134552
    test_log_scope.go:82: use -show-logs to present logs inline
    replicate_queue_test.go:2283: 
        	Error Trace:	pkg/kv/kvserver/replicate_queue_test.go:2283
        	            				pkg/kv/kvserver/replicate_queue_test.go:2308
        	            				pkg/testutils/soon.go:71
        	            				pkg/util/retry/retry.go:355
        	            				pkg/testutils/soon.go:77
        	            				pkg/testutils/soon.go:56
        	            				pkg/testutils/soon.go:38
        	            				pkg/kv/kvserver/replicate_queue_test.go:2307
        	Error:      	Received unexpected error:
        	            	sql: no rows in result set
        	Test:       	TestPromoteNonVoterInAddVoter
    exectrace.go:39: execution trace written to /artifacts/tmp/_tmp/135a86ba3501d232b0a98ae8902ff534/logTestPromoteNonVoterInAddVoter844134552/exectrace_goid_88464.bin
    panic.go:615: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/135a86ba3501d232b0a98ae8902ff534/logTestPromoteNonVoterInAddVoter844134552
--- FAIL: TestPromoteNonVoterInAddVoter (54.94s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestPromoteNonVoterInAddVoter.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>

--- a/pkg/kv/kvserver/replicate_queue_test.go
+++ b/pkg/kv/kvserver/replicate_queue_test.go
@@ -2279,12 +2279,19 @@ SELECT * FROM (
 			return 0, 0, roachpb.RangeDescriptor{}, err
 		}
 
+		// Find the start key of the range containing our table. Note that TABLE t
+		// might not show up in SHOW RANGES immediately. Return zeroes in this case,
+		// and the caller retries.
+		const q = `
+SELECT start_key FROM crdb_internal.ranges_no_leases
+WHERE range_id IN (SELECT range_id FROM [SHOW RANGES FROM TABLE t] LIMIT 1);`
 		var key roachpb.Key
-		require.NoError(t,
-			db.QueryRow(`
-SELECT start_key from crdb_internal.ranges_no_leases WHERE range_id IN
-(SELECT range_id FROM [SHOW RANGES FROM TABLE t] LIMIT 1);
-`).Scan(&key))
+		if err := db.QueryRow(q).Scan(&key); errors.Is(err, gosql.ErrNoRows) {
+			return 0, 0, roachpb.RangeDescriptor{}, nil
+		} else {
+			require.NoError(t, err)
+		}
+
 		desc, err := tc.LookupRange(key)
 		if err != nil {
 			return 0, 0, roachpb.RangeDescriptor{}, err

